src/server/bitops_family.cc

```diff
@@ -0,0 +1,267 @@
+
+OpResult<long> OverRideValue::Set(std::string_view key, int32_t offset, int32_t value) {
+  EngineShard* shard = cmd_.args().shard;
+  auto& db_slice = shard->db_slice();
+
+  DCHECK_LT(params_.db_index, db_slice.db_array_size());
+  DCHECK(db_slice.IsDbValid(params_.db_index));
+
+  auto [it, expire_it] = db_slice.FindExt(
+      params_.db_index, key);  // we need to lookup the value, if exists, we will override it,
+                               // otherwise we will set new one with bit set
+  std::string entry{GetValidNewValue(it, offset, value)};  // existing
+  old_value_ = SetBitValue(entry, offset, value);
+  const auto set_res = cmd_.Set(params_, key, entry);
+  if (set_res == OpStatus::OK) {
+    return OldValue();
+  } else {
+    return set_res;
+  }
+}
+// ------------------------------------------------------------------------- //
+//  Impl for the command functions
+void BitPos(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitCount(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitField(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitFieldRo(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitOp(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendOk();
+}
+
+void GetBit(CmdArgList args, ConnectionContext* cntx) {
+  // Support for the command "GETBIT key offset"
+  // see https://redis.io/commands/getbit/
+
+  int32_t offset{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto cb = [&](Transaction* t, EngineShard* shard) { return ReadValue(t->GetOpArgs(shard), key); };
+  Transaction* trans = cntx->transaction;
+  OpResult<std::string> result = trans->ScheduleSingleHopT(std::move(cb));
+
+  if (result) {
+    DVLOG(2) << "GET" << trans->DebugId() << "': key: '" << key << ", value '" << result.value()
+             << "'\n";
+    // we have the value, now we need to get the bit at the location
+    long val = GetBitValueSafe(result.value(), offset);
+    (*cntx)->SendLong(val);
+  } else {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      default:
+        DVLOG(2) << "GET " << key << " nil";
+        (*cntx)->SendLong(0);  // in case we don't have the value we should just send 0
+    }
+  }
+}
+
+void SetBit(CmdArgList args, ConnectionContext* cntx) {
+  int32_t offset{0};
+  int32_t value{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset) || !absl::SimpleAtoi(ArgS(args, 3), &value)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto index = cntx->db_index();
+  auto memcache_flags = cntx->conn_state.memcache_flag;
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    OverRideValue set_operation{t->GetOpArgs(shard), index, memcache_flags};
```

Comment:
`OpArgs` already provides you with DbIndex so there is no need to pass it as index.
On the other hand - `shard` you need so why not pass it further?
`memcache_flags` you do not need to pass it. memcache_flags is relevant for Memcached commands and SetBit does not exist in Memcached. 

`SET` supports both Redis and Memcached , this is why its code handles memcached_flags.

Reply:
this was change and is out of date